task-sdk/src/airflow/sdk/definitions/weight_rule.py

```diff
@@ -0,0 +1,45 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from enum import Enum
+
+import methodtools
+
+
+# TODO: This can be moved to airflow-shared once it is available
+class WeightRule(str, Enum):
+    """Weight rules."""
+
+    DOWNSTREAM = "downstream"
+    UPSTREAM = "upstream"
+    ABSOLUTE = "absolute"
+
+    @classmethod
+    def is_valid(cls, weight_rule: str) -> bool:
+        """Check if weight rule is valid."""
+        return weight_rule in cls.all_weight_rules()
+
+    @methodtools.lru_cache(maxsize=None)
+    @classmethod
+    def all_weight_rules(cls) -> set[str]:
+        """Return all weight rules."""
+        return set(cls.__members__.values())
+
+    def __str__(self) -> str:
+        return self.value
```

Comment:
Are any of these used inside TaskSDK?

Reply:
Yes it is used indirectly.

Also i have removed this file completely as it is part of `airflow.sdk.api.datamodels._generated` now